src/main/java/randoop/generation/GrtFuzzing.java

```diff
@@ -0,0 +1,480 @@
+      return Math.round(randomGaussian);
+    } else if (cls == float.class || cls == Float.class) {
+      return (float) randomGaussian;
+    } else if (cls == double.class || cls == Double.class) {
+      return randomGaussian;
+    } else {
+      throw new RuntimeException("Unexpected primitive type: " + cls.getName());
+    }
+  }
+
+  /**
+   * Get the methods for fuzzing primitive numbers of the given class.
+   *
+   * @param cls a numeric class
+   * @return a list of methods that will, together, be used to fuzz a number of the given class
+   */
+  private static List<Executable> getNumberFuzzingMethods(Class<?> cls)
+      throws NoSuchMethodException {
+    List<Executable> methodList = new ArrayList<>();
+
+    if (cls == int.class || cls == Integer.class) {
+      methodList.add(Integer.class.getMethod("sum", int.class, int.class));
+    } else if (cls == double.class || cls == Double.class) {
+      methodList.add(Double.class.getMethod("sum", double.class, double.class));
+    } else if (cls == float.class || cls == Float.class) {
+      methodList.add(Float.class.getMethod("sum", float.class, float.class));
+    } else if (cls == long.class || cls == Long.class) {
+      methodList.add(Long.class.getMethod("sum", long.class, long.class));
+    } else if (cls == short.class || cls == Short.class) {
+      // Short doesn't have a sum method, so we use Integer.sum and get the short value
+      methodList.add(Integer.class.getMethod("sum", int.class, int.class));
+      methodList.add(Integer.class.getMethod("valueOf", int.class));
+      methodList.add(Integer.class.getMethod("shortValue"));
+    } else {
+      throw new IllegalArgumentException("Unexpected primitive type: " + cls.getName());
+    }
+
+    return methodList;
+  }
+
+  /**
+   * Create and append all statements needed for the String fuzzing operation to the given sequence.
+   * Precondition: Length of the input String to be fuzzed is not 0.
+   *
+   * @param sequence the (non-empty) sequence to append the String fuzzing operation inputs to
+   * @param operation the String fuzzing operation to perform
+   * @param fuzzStatementCount the number of fuzzing statements added, will be side-effected
+   *     (incremented).
+   * @return a sequence with the String fuzzing operation inputs appended at the end
+   */
+  private static Sequence appendStringFuzzingInputs(
+      Sequence sequence, StringFuzzingOperation operation, AtomicInteger fuzzStatementCount)
+      throws NoSuchMethodException {
+    int stringLength = getStringValue(sequence).length();
+
+    if (stringLength == 0 && operation != StringFuzzingOperation.INSERT) {
+      return sequence; // Cannot remove/replace/substring an empty string
+    }
+
+    sequence = appendStringBuilder(sequence, fuzzStatementCount);
+
+    Sequence fuzzingInputsSequence = getStringFuzzingInputs(operation, stringLength);
+
+    return Sequence.concatenate(sequence, fuzzingInputsSequence);
```

Comment:
This code is a bit inconsistent in that some statements are appended to `sequence` and some are put in `fuzzingInputsSequence`.  Why not put all the extra statements together, and then concatenate just once?